arbnode/batch_poster.go

```diff
@@ -380,6 +382,7 @@ func NewBatchPoster(ctx context.Context, opts *BatchPosterOpts) (*BatchPoster, e
 
 	hotShotUrl := opts.Config().HotShotUrl
 	lightClientAddr := opts.Config().LightClientAddress
+	espressoKeyManagerAddr := opts.Config().EspressoKeyManagerAddress
```

Comment:
Why are we taking this as a config value?

Reply:
We need to generate a new espresso key manager address on start up of the batch poster

Reply:
I believe this is the address of the new pseudo registry contract that the TEE verifier will become in this [PR](https://github.com/EspressoSystems/nitro-contracts/pull/46/files#diff-e69ae1b283d2075cc9e151de8ec995d559d5d2fbe28a908504f89ec90aacb6a1), not a wallet address.

Reply:
There will be no new contract `EspressoTEEVerifier` contract will be the contract, we just have added new functions to that contract

Reply:
Also I saw that later we were using this address to do the signature which seems incorrect